tests/testthat/test_covariance_plot.R

```diff
@@ -84,3 +84,27 @@ test_that("h_vectorization works as expected when string not specified", {
   )
   expect_identical(result, expected)
 })
+
+test_that("g_covarariance works as expected as expected", {
+  vcov_matrix <- matrix(
+    c(49, 12, 12, 23),
+    nrow = 2, ncol = 2,
+    dimnames = list(
+      c(1, 2),
+      c(1, 2)
+    )
+  )
+  result <- g_covariance(vcov_matrix)
+  expected_dataframe <- data.frame(
+    Vect = c(49, 12, 23),
+    time_point_distribution = c(0, 1, 0),
+    lag = c(0, 1, 0),
+    rank_row = c(1, 1, 2),
+    rank_col = c(1, 2, 2)
+  )
+  expected <- ggplot(expected_dataframe, aes(x = lag, y = Vect, colour = rank_row, group = rank_row)) +
+    geom_point() +
+    geom_line() +
+    labs(xlab = "lag", ylab = "")
+  expect_identical(result, expected)
```

Comment:
please use `vdiffr::expect_doppelgaenger()`, see https://github.com/insightsengineering/tern.mmrm/blob/3c569e88990a207393cd2558845b47fe09b75b68/tests/testthat/test-g_mmrm.R#L17 for an example

Reply:
Thank you @danielinteractive. I have added the unit tests but I am a little confused / curious on how this tests exactly when you plug result into `vdiffr::expect_doppelganger()`

Reply:
Honestly I don't know the internals, you can find more documentation here: https://vdiffr.r-lib.org/reference/expect_doppelganger.html